op_crates/web/00_webidl.js

```diff
@@ -0,0 +1,668 @@
+    const x = toNumber(V, opts);
+
+    if (isNaN(x)) {
+      return x;
+    }
+
+    if (Object.is(x, -0)) {
+      return x;
+    }
+
+    return Math.fround(x);
+  };
+
+  converters.DOMString = function (V, opts = {}) {
+    if (opts.treatNullAsEmptyString && V === null) {
+      return "";
+    }
+
+    if (typeof V === "symbol") {
+      throw makeException(
+        TypeError,
+        "is a symbol, which cannot be converted to a string",
+        opts,
+      );
+    }
+
+    const StringCtor = opts.globals ? opts.globals.String : String;
+    return StringCtor(V);
+  };
+
+  converters.ByteString = (V, opts) => {
+    const x = converters.DOMString(V, opts);
+    let c;
+    for (let i = 0; (c = x.codePointAt(i)) !== undefined; ++i) {
+      if (c > 255) {
+        throw makeException(TypeError, "is not a valid ByteString", opts);
+      }
+    }
+
+    return x;
+  };
+
+  converters.USVString = (V, opts) => {
+    const S = converters.DOMString(V, opts);
+    const n = S.length;
+    const U = [];
+    for (let i = 0; i < n; ++i) {
+      const c = S.charCodeAt(i);
+      if (c < 0xD800 || c > 0xDFFF) {
+        U.push(String.fromCodePoint(c));
+      } else if (0xDC00 <= c && c <= 0xDFFF) {
+        U.push(String.fromCodePoint(0xFFFD));
+      } else if (i === n - 1) {
+        U.push(String.fromCodePoint(0xFFFD));
+      } else {
+        const d = S.charCodeAt(i + 1);
+        if (0xDC00 <= d && d <= 0xDFFF) {
+          const a = c & 0x3FF;
+          const b = d & 0x3FF;
+          U.push(String.fromCodePoint((2 << 15) + ((2 << 9) * a) + b));
+          ++i;
+        } else {
+          U.push(String.fromCodePoint(0xFFFD));
+        }
+      }
+    }
+
+    return U.join("");
+  };
+
+  converters.object = (V, opts) => {
+    if (type(V) !== "Object") {
+      throw makeException(TypeError, "is not an object", opts);
+    }
+
+    return V;
+  };
+
+  // Not exported, but used in Function and VoidFunction.
+
+  // Neither Function nor VoidFunction is defined with [TreatNonObjectAsNull], so
+  // handling for that is omitted.
+  function convertCallbackFunction(V, opts) {
+    if (typeof V !== "function") {
+      throw makeException(TypeError, "is not a function", opts);
+    }
+    return V;
+  }
+
+  const abByteLengthGetter =
+    Object.getOwnPropertyDescriptor(ArrayBuffer.prototype, "byteLength").get;
+
+  function isNonSharedArrayBuffer(V) {
+    try {
+      // This will throw on SharedArrayBuffers, but not detached ArrayBuffers.
+      // (The spec says it should throw, but the spec conflicts with implementations: https://github.com/tc39/ecma262/issues/678)
+      abByteLengthGetter.call(V);
+
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  function isSharedArrayBuffer(V) {
+    // TODO(lucacasonat): voulnrable to prototype polution. Needs to happen
+    // here because SharedArrayBuffer is not available during snapshotting.
+    const sabByteLengthGetter =
```

Comment:
Consider having an outer `let` variable that you memoize with the prototype once.  While it won't avoid prototype pollution totally, you only bind once instead every call to the method.